src/lib.rs

```diff
@@ -88,13 +99,23 @@ pub enum Block {
     HorizontalRule,
     /// Table, with caption, column alignments (required), relative column widths (0 = default),
     /// column headers (each a list of blocks), and rows (each a list of lists of blocks)
+    #[cfg(feature = "simple_table")]
     Table(
         Vec<Inline>,
         Vec<Alignment>,
         Vec<Double>,
         Vec<TableCell>,
         Vec<Vec<TableCell>>,
     ),
+    #[cfg(feature = "new_table")]
```

Comment:
Cargo features are meant to be additive, so it should be allowed to enable both features and everything still works.

That said, I know that lots of crates use these features this way and it's probably very unlikely you'll end up with both features without also being a binary crate, and thus having full control over everything.

Maybe it could be better to just do a major version bump and do this change there? What do you think?

Reply:
Alternatively we can rename the new table enum variant as `NewTable` and keep the old one, even if it is never emitted. Then we do not need a `simple_table` feature and can just differentiate between `not(feature = "new_table")` and `feature = "new_table")` in the runtime code

Reply:
I know it is a bit "hacky" and not the "rusty way" of dealing with features. There were only two options I could think of that are backwards compatible: selection at compile time (this way) and at runtime (which needs a custom `Deserialize` implementation to choose between `Table` and `NewTable`).

As you suggested, introducing changes in a new major version could be better, as the new table format is a breaking change in pandoc's AST. The question is whether support for the old table should remain in the new version or not (that is what [panflute](https://github.com/sergiocorreia/panflute/tree/ce9709f84da144d77d5e447186c10971fa284c75#supported-pandoc-versions) did in their 2.0 version a weeks days ago).

Reply:
I'm fine dropping support for the old table